### DISCUSSION TITLE
Multi-objective optimisation?

### DISCUSSION THREAD

**Discussion by MichaelClerx** [pints-team/pints#1642] [Ideas/Science]
Should we add something for MOO at some point?

(Optimisation with multiple (incompatible) criteria, where the goal is to find a solution (or the set of solutions) such that improvement of any one of the criteria will make the others worse)

Not sure how these would relate to sampling

**Comment by ben18785** [comment]
Sounds interesting! This is similar to a concept in economics called Pareto
Optimality (https://en.wikipedia.org/wiki/Pareto_efficiency).

**Comment by MichaelClerx** [comment]
Yeah it's the same :D 

Just having a look at this paper

  https://doi.org/10.1093/bib/bby092

on calibrating biological models, which talks about MOO a bit, and links to an algorithm for it

  https://github.com/marknormanread/unsga3

We did this stuff in ML at some point too. But can't help but thinking that if your objectives are incompatible in biological modelling (as opposed to some economics / policy scenario) maybe you've just not got the physics right? Although you could still have a case where you have the same underlying physical model for the biology, but a different noise model for the experiment, which could lead to this situation?
